### ISSUE TITLE
How to create web proxy for Api ?

### ISSUE THREAD

**Issue by jaskaranbhatti** [justcoding121/Titanium-Web-Proxy#700]
There is existing web api hosted, l am trying to create application so that in my application when i call Api it will eventually hit the proxy instead of calling API. Also this Api is hosted on IBM APIM with added security like certificates, etc. Can we create proxy for such cases ?

**Comment by honfika** [comment]
Yes, I think you can create it if you can set a proxy address for your webapi requests.

**Comment by justcoding121** [comment]
Closing this as a usage question. Start an ExplicitProxyEndPoint and configure the HTTP client making the API call to use it:

```csharp
var endPoint = new ExplicitProxyEndPoint(IPAddress.Loopback, 8000);
proxyServer.AddEndPoint(endPoint);
proxyServer.Start();

var handler = new HttpClientHandler
{
    Proxy = new WebProxy(http://127.0.0.1:8000),
    UseProxy = true
};
var client = new HttpClient(handler);
```

Token headers pass through normally. If the API requires mutual TLS, handle ClientCertificateSelectionCallback so Titanium can present the required client certificate upstream. The Titanium root certificate must also be trusted by the client when HTTPS decryption is enabled.
